File: Backend/parse_and_inject.js
const fs = require('fs');
const path = require('path');

const dataFile = 'C:\\Users\\Asus\\.gemini\\antigravity-ide\\brain\\1810db71-6e9e-49fd-9cfa-5a33ca325207\\scratch\\extracted_data_utf8.jsonl';
const dbPath = 'C:\\Users\\Asus\\Desktop\\NowicME\\NowicSTDO\\Backend\\dev.sqlite3';

// Icon mapping helper
function getRandomIcon() {
    const icons = ['Rocket', 'Bot', 'Layers', 'Gauge', 'Cpu', 'Code2', 'Sparkles', 'Globe', 'Zap', 'Trophy', 'Users', 'Star', 'ShieldCheck'];
    return icons[Math.floor(Math.random() * icons.length)];
}

async function run() {
    const lines = fs.readFileSync(dataFile, 'utf8').split('\n');
    let contentStr = '';
    for (const line of lines) {
        if (!line.trim()) continue;
        try {
            const parsed = JSON.parse(line);
            if (parsed.type === 'USER_INPUT' && parsed.content.includes('Detailed Services Raw Data Extraction')) {
                contentStr = parsed.content;
                break;
            }
        } catch(e) {}
    }

    if (!contentStr) {
        console.error('Could not find content in jsonl');
        return;
    }

    const serviceBlocks = contentStr.split('Service: ').slice(1);
    const mappedServices = [];

    for (let block of serviceBlocks) {
        const lines = block.split('\n');
        const slug = lines[0].trim();
        
        let code = lines.slice(1).join('\n');
        
        // Remove markdown tags if any
        code = code.replace(/```javascript/g, '').replace(/```/g, '');
        
        // Replace `icon: ,` or `icon: Cpu,` with `icon: "Cpu",`
        code = code.replace(/icon:\s*([a-zA-Z0-9_]+)?,/g, (match, p1) => {
            if (!p1) return `icon: "${getRandomIcon()}",`;
            return `icon: "${p1}",`;
        });
        code = code.replace(/icon:\s*([a-zA-Z0-9_]+)?\s*\}/g, (match, p1) => {
            if (!p1) return `icon: "${getRandomIcon()}" }`;
            return `icon: "${p1}" }`;
        });

        // Remove any line that is just 'n' or doesn't have a const definition at the top level
        code = code.replace(/^n$/gm, '');
        if (code.includes('const faqs =')) {
            const parts = code.split('const faqs =');
            code = parts[0] + 'const faqs =' + parts[1].split('];')[0] + '];';
        }
        
        // Add dummy exports so we can eval it
        code += `\n return { heroContent, introduction: (typeof introduction !== 'undefined' ? introduction : ''), services: (typeof webServices !== 'undefined' ? webServices : typeof appServices !== 'undefined' ? appServices : typeof customSoftwareServices !== 'undefined' ? customSoftwareServices : typeof marketingServices !== 'undefined' ? marketingServices : typeof designServices !== 'undefined' ? designServices : typeof aiServices !== 'undefined' ? aiServices : []), process: (typeof developmentProcess !== 'undefined' ? developmentProcess : typeof marketingProcess !== 'undefined' ? marketingProcess : typeof designProcess !== 'undefined' ? designProcess : []), whyChooseUsData, faqs };`;
        
        try {
            const func = new Function(code);
            const data = func();
            
            // Map keys
            const finalService = {
                slug: slug,
                heroContent: data.heroContent,
                introduction: data.introduction,
                subServices: data.services.map(s => ({
                    title: s.title,
                    description: s.description,
                    icon: s.icon,
                    features: s.keyFeatures || s.tools || []
                })),
                process: data.process.map(p => ({
                    step: p.number,
                    title: p.title,
                    description: p.description
                })),
                whyChooseUs: data.whyChooseUsData.map(w => ({
                    icon: w.icon,
                    title: w.title,
                    description: w.description
                })),
                faqs: data.faqs.map(f => ({
                    q: f.question,
                    a: f.answer
                }))
            };
            mappedServices.push(finalService);
        } catch (e) {
            console.error('Failed to parse block for', slug, e);
        }
    }

    console.log(`Parsed ${mappedServices.length} services.`);
    
    fs.writeFileSync('C:\\Users\\Asus\\Desktop\\NowicME\\NowicSTDO\\Backend\\mapped_services.json', JSON.stringify(mappedServices, null, 2));
    console.log('Saved mapped_services.json');
}

run();

File: Backend/parse_and_inject.py
import json
import sqlite3
import re
import random

data_file = r'C:\Users\Asus\.gemini\antigravity-ide\brain\1810db71-6e9e-49fd-9cfa-5a33ca325207\scratch\extracted_data_utf8.jsonl'
db_path = r'C:\Users\Asus\Desktop\NowicME\NowicSTDO\Backend\dev.sqlite3'

icons = ['Rocket', 'Bot', 'Layers', 'Gauge', 'Cpu', 'Code2', 'Sparkles', 'Globe', 'Zap', 'Trophy', 'Users', 'Star', 'ShieldCheck']

def get_random_icon():
    return random.choice(icons)

def run():
    with open(data_file, 'r', encoding='utf-8') as f:
        content_str = ''
        for line in f:
            if not line.strip(): continue
            try:
                parsed = json.loads(line)
                if parsed.get('type') == 'USER_INPUT' and 'Detailed Services Raw Data Extraction' in parsed.get('content', ''):
                    content_str = parsed['content']
                    break
            except Exception:
                pass

    if not content_str:
        print("Could not find content in jsonl")
        return

    service_blocks = content_str.split('Service: ')[1:]
    
    # We will use quick regex to extract the properties
    mapped_services = []

    for block in service_blocks:
        lines = block.split('\n')
        slug = lines[0].strip()
        
        # Regex to find JSON-like objects in the code
        # Because the user provided it as JS code, Python regex is easiest for extraction.
        # But honestly, we can just write a quick JS script and run it in a node context where we install sqlite3?
        # Actually it's faster to run it with node by running `npm install sqlite3` in a temp folder.
        pass

if __name__ == '__main__':
    run()
